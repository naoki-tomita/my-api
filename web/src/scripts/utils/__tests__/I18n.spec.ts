import { t, I18n } from "../I18n";

describe("I18n", () => {
  describe("#init", () => {
    it("should fetch resource", async () => {
      const i18n = new I18n();
      expect(await i18n.init("en")).toBe(t);
      expect(i18n.map).toMatchSnapshot();
    });
  });

  describe("#t", () => {
    const tests: { map: any, key: string, expected: string }[] = [
      {
        map: { foo: "bar" },
        key: "foo",
        expected: "bar",
      },
      {
        map: { foo: { bar: "baz" } },
        key: "foo.bar",
        expected: "baz",
      },
      {
        map: { foo: { foo: "baz" } },
        key: "foo.bar",
        expected: "foo.bar",
      },
      {
        map: {},
        key: "foo.bar",
        expected: "foo.bar",
      },
      {
        map: { foo: { bar: { baz: "baz" }} },
        key: "foo.bar",
        expected: "foo.bar",
      },
      {
        map: null,
        key: "foo.bar",
        expected: "foo.bar",
      },
    ];

    tests.forEach(({ map, key, expected }) => {
      it("should pass the test.", () => {
        const i18n = new I18n();
        i18n.map = map;
        expect(i18n.t(key)).toBe(expected);
      });
    });

    it("should replace __foo__ with text.", () => {
      const i18n = new I18n();
      i18n.map = { foo: "__foo__ bar" };
      expect(i18n.t("foo", { foo: "bar" })).toBe("bar bar");
    });
  });
});
