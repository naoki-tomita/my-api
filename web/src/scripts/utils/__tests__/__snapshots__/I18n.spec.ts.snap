// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`I18n #init should fetch resource 1`] = `
Object {
  "default": Object {
    "profile": Object {
      "description": "Self introduction",
      "displayName": "Name",
    },
  },
  "profile": Object {
    "description": "Self introduction",
    "displayName": "Name",
  },
}
`;
